File: Sources/Engine/Core/Math/Vec/Vec4.cpp
#include "Chicane/Core/Math/Vec/Vec4.reflected.hpp"

namespace Chicane
{
    const Vec4 Vec4::Zero = Vec4(0.0f);
    const Vec4 Vec4::One  = Vec4(1.0f);

    const Vec4 Vec4::Right   = Vec4(1.0f, 0.0f, 0.0f, 0.0f);
    const Vec4 Vec4::Forward = Vec4(0.0f, 1.0f, 0.0f, 0.0f);
    const Vec4 Vec4::Up      = Vec4(0.0f, 0.0f, 1.0f, 0.0f);

    String Vec4::toString() const
    {
        return String::sprint("[%.2f, %.2f, %.2f, %2f]", x, y, z, w);
    }
}

File: Includes/Chicane/Core/Script/Types/Vec4.lua
---@meta

---@class Vec4
---@field x number
---@field y number
---@field z number
---@field w number
Vec4 = {}

---@param x number
---@param y number
---@param z number
---@param w number
---@return Vec4
function Vec4(x, y, z, w) end

---@return Vec4
function Vec4.Zero() end

---@return Vec4
function Vec4.One() end

---@return Vec4
function Vec4.Right() end

---@return Vec4
function Vec4.Forward() end

---@return Vec4
function Vec4.Up() end

---@return Vec4
function Vec4.Sentinel() end

---@return string
function Vec4:toString() end
